### ISSUE TITLE
Date Calculation Does Not Factor In Leap Year

### ISSUE THREAD

**Issue by MicrosoftIssueBot** [microsoft/calculator#53]
**Describe the bug**
The next occurring leap year is next year 2020! The actual Leap Day is Saturday Feburary 29th, 2020. In a recent round of tests with the Date Calculation I found that when adding days past the next leap day the calculation acted as if that day never existed and didn't account for the fact there is an extra day in the year. I set up the case with Today's date February 20,2019 and added 1 year,  0 months and 6 days. The answer came to be A February 26, 2020 just 3 days before Leap Day. I then added an extra year to the equation and got February 26, 2021, which normally that is fine but with leap year it would actually be February 25th, 2021 because of the extra day.

**Steps To Reproduce**
1. Launch the Calculator app build 10.1812.10048.0.
2. Navigate to Date Calculation.
3. From the drop down field select Add or subtract days.
4. For the from date enter February 20,2019
5. Check the bubble Add
6. Enter 1 Year, 0 Months, 6 Days.
7. Take Note of the date Wednesday, February 26,2020 (3 days before Leap Day)
8. Change Years field from 1 to 2.
9. Observe new date Friday, February 26, 2021. 

**Expected behavior**
Leap years are taken into account for date calculations.

**Device and Application Information (please complete the following information):**
 - OS Build Number:
 - Architecture: x64
 - Application Version Number: 10.1812.10048.0

 

**Additional context**
What is also interesting here is that normally the days of the week move forward one day from the previous, except in the case of leap year and then it is two for any date past Leap Day. So looking at this the days of the week has been adjusted for leap year, but the calendar date was not.

**Comment by benjaminkwilliams** [comment]
I believe that test case is incorrect. Per https://www.timeanddate.com/date/dateadded.html?m1=02&d1=20&y1=2019&type=add&ay=2&am=&aw=&ad=6&rec= the correct value was returned. When you add years, you do not convert them to days, you just add years. Does it reproduce when starting with Feb-1 and adding 31 days or Feb-28 00:00 and adding 25 hours?

**Comment by marcomontoya** [comment]
I agree with @benjaminkwilliams. I think it should be asked what does the user expect when a year is added to a date, if the next day in the next year or the 365th next day. I find it more likely to be the first one. Otherwise, the answer for adding y years to a date month/day/year would be month/day - leap years in that period o time/year+y, and saying that 20 years after 02/20/2019 is 02/15/2039 sounds odd to me.
However, if we stick to the definition of adding a year that Benjamin and I propose, it would be still needed to be define what date is a year after any February 29th.

**Comment by rudyhuyn** [comment]
I agree with @Benjaminkwilliams and @marcomontoya, a year is not defined as 365 days neither as Earth doing a full loop around the sun (~365.24219 days), but as the same the same day number and month but Year+1. 

We should also keep in mind that the Calculator application manages more than the Gregorian calendar.

Think about birthdays or anniversaries, you don't celebrate your birthday 1 day before before the current year was a leap one.

**Comment by mj1856** [comment]
I concur, the calculator is behaving correctly.  It is indeed taking leap years into account, by *not* treating a year as 365 days.  This is akin to not every month being 30 days.

In both cases, the calculator is doing the correct thing.
- Consider March 31st (of any year) plus one month.  The result cannot be April 31st, because April only has 30 days.  So the result becomes April 30th.
  - Advancing to May 1st would not make sense in most cases - as we started in March so expected to end in April.
- Now consider February 29th 2020 plus one year.  The result cannot be February 29th 2021, because in that year February only has 28 days.  Thus, the result is February 28th 2021.
  - Advancing to March 1st 2021 would not make sense in most cases - as we started in February so expected to end in February.

The issue should be closed.  Thanks.
